packages/effective-module-tree/index.js

```diff
@@ -0,0 +1,109 @@
+const fs = require( 'fs' ).promises;
+const path = require( 'path' );
+const globby = require( 'globby' );
+const debug = require( 'debug' )( 'effective-module-tree' );
+const treeify = require( 'object-treeify' );
+
+// Generator that yields all places were node will look for modules
+// For a path like : a/node_modules/b/node_modules/c
+// It will generate the candidates:
+//		- a/node_modules/b/node_modules/c/node_modules
+// 		- a/node_modules/b/node_modules
+//		- a/node_modules
+//		- node_modules
+const candidates = function*( packagePath ) {
+	const parts = packagePath.split( path.sep );
+	for ( let i = parts.length; i >= 0; i-- ) {
+		// This avoid generating .../node_modules/node_modules/...
+		if ( parts[ i - 1 ] === 'node_modules' ) continue;
+		yield path.join( ...parts.slice( 0, i ), 'node_modules' );
+	}
+};
+
+// Simplify tree. Once the full tree has been discovered and linked, we need
+// to simplify it (i.e. remove unused properties) to generate a format compatible
+// with treeify so we can render it (https://www.npmjs.com/package/object-treeify)
+const simplify = ( packageDef, parents ) => {
```

Comment:
"simplify" and "remove unused properties" are so vague terms 🙂 What does the function _really_ do?